### PR TITLE
[ws-daemon] Fix workspace location for log backup

### PR DESCRIPTION
**Pull request by WVerlaek**
## Description

Fix workspace location for logs backup. Use the session's location (path on daemon host) instead of the WorkspaceLocation of the CRD (comes from .gitpod.yml's workspaceLocation)

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 4c9fa34</samp>

Refactor `uploadWorkspaceLogs` and `BackupOptions` to use workspace location from `Workspace` object. This reduces code complexity and redundancy.

</details>

## Related Issue(s)

Fixes ENG-50

## How to test

Tested in a preview env, by creating a prebuild without this change (no logs in dashboard when viewing the completed prebuild), and then a prebuild after this change, which does show logs even after reloading the page.

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Review by kylos101 — APPROVED** [review]
:+1: 

Tested we can snapshot and restore, too.

Tested prebuilds stream while prebuilding, can start from a prebuild, stop, and restart

Tested backup of a regular workspace, and related restart

File: components/ws-daemon/pkg/controller/workspace_operations.go [hunk 1]
```diff
@@ -345,9 +344,9 @@ func ensureCleanSlate(location string) error {
 	return nil
 }
 
-func (wso *DefaultWorkspaceOperations) uploadWorkspaceLogs(ctx context.Context, opts BackupOptions) (err error) {
+func (wso *DefaultWorkspaceOperations) uploadWorkspaceLogs(ctx context.Context, opts BackupOptions, location string) (err error) {
```
Comment (easyCZ) [inline components/ws-daemon/pkg/controller/workspace_operations.go:347]: Any unit tests we need to add/update as part of this to ensure the regression doesn't happen again in the future?

**Review by Furisto — APPROVED** [review]

File: components/ws-daemon/pkg/controller/workspace_operations.go (hunk 1 above)
Comment (WVerlaek) [reply components/ws-daemon/pkg/controller/workspace_operations.go:347]: I've updated an existing integration test to check for uploaded logs. There's no unit tests nor is it easy to add one now that would catch this regression

**Review by geropl — APPROVED** [review]
:heavy_check_mark:

**Pull request merged**